sdk/typescript/src/signers/txn-data-serializers/txn-data-serializer.ts

```diff
@@ -62,8 +62,7 @@ export interface MoveCallTransaction {
   gasBudget: number;
 }
 
-/** A type that represents the possible transactions that can be signed: */
-export type SignableTransaction =
+export type UnserializedSignableTransaction =
```

Comment:
`UnserializedSignableTransaction` all have `gasPayment` fields. Having a separate type for this can avoid the need to check whether `gasPayment` field exists or not:

https://github.com/MystenLabs/sui/blob/87e1314ef61fc39904a612bcf9d96481065f02bb/sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts#L306